### PR TITLE
Replace epoll with select

### LINKED ISSUE #5: Support OSX by using alternative to epoll()
I came here through https://github.com/cs01/gdbgui/ which looks brilliant.

My problem is that I do all my Rust development in OSX, which does not support epoll().
https://github.com/cs01/pygdbmi/blob/master/pygdbmi/gdbcontroller.py#L79

Maybe have a look at this file from py-prefork-server?
https://github.com/crustymonkey/py-prefork-server/blob/ff9c1411f27843c6384d6c5b2dfcc2c09d4642bf/preforkserver/poller.py

### PR DESCRIPTION
**Pull request by felipesere**
`make test` now passes on my Mac

Closes #5

### CODE REVIEW

**Comment by rakeshmane** [comment]
May be you forgot to remove line number 17 to 22 :
    select.POLLIN: 'POLLIN',
    select.POLLPRI: 'POLLPRI',
    select.POLLOUT: 'POLLOUT',
    select.POLLERR: 'POLLERR',
    select.POLLHUP: 'POLLHUP',
    select.POLLNVAL: 'POLLNVAL'

**Comment by cs01** [comment]
Have you tested that `gdbgui` can run with this change?

**Comment by felipesere** [comment]
I haven't. How do I point `gdbgui` to my local `pygdbmi`?

**Comment by rakeshmane** [comment]
@cs01 I can confirm that now it's working fine on my mac.

**Comment by rakeshmane** [comment]
@felipesere You don't need to. 
Just install your fixed local version of pygdbmi first then install gdbgui using pip : pip install gdbgui

**Comment by felipesere** [comment]
Works 👍

File: pygdbmi/gdbcontroller.py [hunk 1]
```diff
@@ -75,11 +63,8 @@ def __init__(self, gdb_path='gdb', gdb_args=['--nx', '--quiet', '--interpreter=m
         self.stderr_fileno = self.gdb_process.stderr.fileno()
         self.stdin_fileno = self.gdb_process.stdin.fileno()
 
-        # register stdout and stderr for epoll
-        self.epoll.register(self.gdb_process.stdout.fileno(), select.EPOLLIN)  # There is data to read
-        self.epoll.register(self.gdb_process.stderr.fileno(), select.EPOLLIN)  # There is data to read
-        self.epoll_write.register(self.gdb_process.stdin.fileno(), select.POLLOUT)  # Ready for output: writing will not block
-
+        self.read_list = [self.stdin_fileno, self.stdout_fileno, self.stderr_fileno]
```
Comment (cs01) [inline pygdbmi/gdbcontroller.py]: why is stdin in the read list?

Comment (felipesere) [inline pygdbmi/gdbcontroller.py]: Good catch.  It's not supposed to be dupicated with the `write_list`.

**Comment by rakeshmane** [comment]
May be there's something wrong with the fix, I am not able to debug anything.
When I try to run executable in gdbgui, it shows : 

 `Unable to find Mach task port for process-id 7270: (os/kern) failure (0x5).\n (please check gdb is codesigned - see taskgated(8))`

**Comment by felipesere** [comment]
I got the same, but I thought that was me trying to debug a Rust application for the first time...
Googling around, it seems that `gdb` needs to be "codesigned" before you can use it on Mac:
https://opensource.apple.com/source/lldb/lldb-69/docs/code-signing.txt

**Comment by cs01** [comment]
Yeah that is an annoyance for OSX. Is there any way either of you could provide a friendly error from pygdbmi and point to a link on how to fix the issue?

**Pull request merged**